draft-ietf-tls-esni.md

```diff
@@ -429,6 +429,18 @@ The "ech_outer_extensions" extension is only used for compressing the
 ClientHelloInner. It MUST NOT be sent in either ClientHelloOuter or
 ClientHelloInner.
 
+Note that it is possible to implement decoding of the EncodedClientHelloInner in
+a way that creates a denial-of-service vulnerability. Specifically, the server
+needs to check that each extension in the OuterExtensions list appears in the
+ClientHelloOuter. The naive strategy would require N\*M comparisons in the worst
+case, where N is the number of extensions in the ClientHelloOuter and M is the
+number of extensions in the OuterExtensions list. Malicious clients could
+exploit this behavior in order to degrade the performance or availability of the
+server. This problem can be mitigated by representing OuterExtensions in a way
```

Comment:
```suggestion
exploit this behavior in order to cause excessive work for the server.
This problem can be mitigated by representing OuterExtensions in a way
```

Reply:
Done, I also added ", possibly making it unavailable." (Worth mentioning since it's the goal of the attacker.)